code/modules/research/protolathe.dm

```diff
@@ -202,3 +202,16 @@
 			if(new_item.matter && new_item.matter.len > 0)
 				for(var/i in new_item.matter)
 					new_item.matter[i] = new_item.matter[i] * mat_efficiency
+
+/obj/machinery/r_n_d/protolathe/proc/eject(var/material, var/amount)

```

Comment:
``r_n_d`` doesn't have a materials var. Ideally there should be something like ``r_n_d/constructor`` where ``constructor`` would have all the common properties - or, better, unify CI, both lathes, and mechfab, but that's a pretty distant goal and the fix is needed. There are the same for CI and lathe.